src/Command.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Equip\Command;
+
+/**
+ * A general purpose command class.
+ *
+ * @since 1.3.0
+ */
+abstract class Command
+{
+    /**
+     * @var OptionsInterface
+     */
+    private $options;
+
+    /**
+     * Execute the command using the current options.
+     *
+     * @return mixed
+     */
+    abstract public function execute();
+
+    /**
+     * Allow usage as a callable.
+     *
+     * @see Command::execute()
+     *
+     * @return mixed
+     */
+    final public function __invoke()
+    {
+        return $this->execute();
+    }
+
+    /**
+     * Get the currently defined options.
+     *
+     * @return OptionsInterface
+     *
+     * @throws CommandException
+     *  If no options have been added to the command.
+     */
+    final public function options()
+    {
+        if (!$this->options) {
+            throw CommandException::needsOptions($this);
+        }
+
+        return $this->options;
+    }
+
+    /**
+     * Get a copy with new options.
+     *
+     * @param Options $options
```

Comment:
Corrected this type hint, as per [this Scrutinizer issue](https://scrutinizer-ci.com/g/equip/command/inspections/fd08a0c3-1f33-43ae-afdc-a229929068da/issues/files/src/Command.php?status=new&orderField=path&order=asc&honorSelectedPaths=0).